### PR TITLE
SERVICE: Password hasher

### LINKED ISSUE #85: SERVISE: COMMON/SECURITY/PasswordHasher Create

### PR DESCRIPTION
**Pull request by SamandarYigitaliyev**
close #85

### CODE REVIEW

File: src/GreenSaleuz.Persistence/Validators/PasswordValidator.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            if (char.IsUpper(item)) isUpperCaseExists = true;
+            if (char.IsLower(item)) isLowerCaseExists = true;
+            if (char.IsDigit(item)) isNumberExists = true;
+            if (Symbols.Contains(item)) isCharacterExists = true;
+        }
+
+        if (isNumberExists == false) return (IsValid: false, Message: "Password should contain at least one Digit!");
+        if (isUpperCaseExists == false) return (IsValid: false, Message: "Password should contain at least one Upper case!");
+        if (isLowerCaseExists == false) return (IsValid: false, Message: "Password should contain at least one Lower case!");
+        if (isCharacterExists == false) return (IsValid: false, Message: "Password should contain at least one Symbol like (#@$%.!)!");
```
Comment (rustambekovich) [inline src/GreenSaleuz.Persistence/Validators/PasswordValidator.cs:27]: A lot of charackter 130 ++

File: src/GreenSaleuz.Persistence/Validators/PhoneNumberValidator.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        if(phoneNumber.Length!=13) return false;
+
+        if(phoneNumber.StartsWith("+998") == false) return false;
+
+        for (int i = 1; i <phoneNumber.Length; i++)
+        {
+            if (char.IsDigit(phoneNumber[i]) == true) continue;
+            else return false;
+        }
+
```
Comment (rustambekovich) [inline src/GreenSaleuz.Persistence/Validators/PhoneNumberValidator.cs:16]: not empty lineor drop line

File: src/GreenSaleuz.Service/Common/Security/PasswordHasher.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+public class PasswordHasher
+{
+    public static (string Hash, string Salt) Hash(string password)
+    {
+        string salt = Guid.NewGuid().ToString();
+        string hash = BCrypt.Net.BCrypt.HashPassword(password + salt);
+
+        return (Hash: hash, Salt: salt);
+    }
```
Comment (rustambekovich) [inline src/GreenSaleuz.Service/Common/Security/PasswordHasher.cs:11]: you need yo 1 empty line

Comment (SamandarYigitaliyev) [reply src/GreenSaleuz.Service/Common/Security/PasswordHasher.cs:11]: Ozi bitta turbdiku

**Pull request merged**